### PR TITLE
Fix -Wall warnings

### PR DESCRIPTION
**Pull request by tom-bop**

### CODE REVIEW

File: abstract-par/Control/Monad/Par/Class.hs [hunk 1]
```diff
@@ -144,26 +144,28 @@ class ParFuture ivar m  => ParIVar ivar m | m -> ivar where
 --   in the /local/ @Par@ thread.  That is, at @fork@ points it is
 --   necessary to give the child computation a separate set of stream
 --   cursors so that it observes the same sequences as the parent.
+{-
```
Comment (simonmar) [inline abstract-par/Control/Monad/Par/Class.hs]: Is this not used anywhere? We probably shouldn't have commented-out code.

File: monad-par-extras/Control/Monad/Par/State.hs [hunk 2]
```diff
@@ -61,7 +60,7 @@ instance (SplittableState s, PC.ParIVar iv p)
               do s <- S.get
                  let (s1,s2) = splitState s
                  S.put s2
-                 lift$ PC.fork $ do S.runStateT task s1; return ()
+                 lift$ PC.fork $ do _ <- S.runStateT task s1; return ()
```
Comment (simonmar) [inline monad-par-extras/Control/Monad/Par/State.hs]: Meh, I'm not a fan of that warning, we could also turn it off.

File: monad-par/Control/Monad/Par/Scheds/Direct.hs [hunk 3]
```diff
@@ -34,12 +34,12 @@ module Control.Monad.Par.Scheds.Direct (
...
 import Control.Concurrent hiding (yield)
 import Data.IORef         (IORef,newIORef,readIORef,writeIORef,atomicModifyIORef)
-import Text.Printf        (printf, hPrintf)
+import Text.Printf        (printf)
 import GHC.Conc           (numCapabilities,yield)
 import           "mtl" Control.Monad.Cont as C
 import qualified "mtl" Control.Monad.Reader as RD
 import qualified       System.Random.MWC as Random
-import                 System.IO  (stderr)
+-- import                 System.IO  (stderr)
```
Comment (simonmar) [inline monad-par/Control/Monad/Par/Scheds/Direct.hs]: Don't comment out code

File: abstract-par/Control/Monad/Par/Class.hs (hunk 1 above)
Comment (tom-bop) [reply abstract-par/Control/Monad/Par/Class.hs]: The only reference is to `ParChan not released yet` from 2012. (https://github.com/simonmar/monad-par/blob/master/monad-par-extras/Control/Monad/Par/State.hs#L71). Should I remove it, or add a `#if 0` like in the other module?

Comment (tom-bop) [reply abstract-par/Control/Monad/Par/Class.hs]: It's also not exported by the module https://github.com/simonmar/monad-par/blob/master/abstract-par/Control/Monad/Par/Class.hs#L33 (also from 2012)

**Pull request merged**